scripts/docs.js

```diff
@@ -0,0 +1,144 @@
+  return `[\`${name}\`](#${name})`;
+};
+
+const getPath = obj => {
+  if (obj.parent === null || obj.parent.name === `"${pathToDeclaration}"`)
+    return obj.name;
+  return `${getPath(obj.parent)}.${obj.name}`;
+};
+
+const getMarkdown = (obj, parent = null) => {
+  const { kind } = obj;
+  /* FunctionDeclaration */
+  if (kind === 216) {
+    const { type } = obj;
+    const typeName = type.slice(type.indexOf('<') + 1, type.lastIndexOf('>'));
+    const typeHeading = typeName.split('.').pop();
+    return ''.concat(
+      `### \`${getPath(parent)}.${obj.name}(${obj.parameters
+        .map(param => param.name)
+        .join(', ')})\`\n\n`,
+      `${obj.documentation}\n\n`,
+      `Expects the following parameters:\n\n${obj.parameters.reduce(
+        (acc, val) => acc.concat(`- ${val.name} (${getType(val.type)})\n\n`),
+        ''
+      )}`,
+      `Returns [${type}](#${typeHeading})`,
+      '\n\n'
+    );
+  }
+  /* ClassDeclaration */
+  if (kind === 217) {
+    let md = `## Class <a name="${obj.name}">${obj.name}</a>\n\n`;
+    const constructor = obj.members.shift().parameters[0];
+    md += `Constructed with:\n- ${constructor.name} (\`${constructor.type}\`)\n`;
+    if (obj.members) {
+      obj.members.forEach(child => {
+        md = md.concat(getMarkdown(child));
+      });
+    }
+    return md;
+  }
+  /* InterfaceDeclaration */
+  if (kind === 218) {
+    return ''.concat(
+      `### <a name="${obj.name}">${obj.name}</a>\n\n`,
+      `Object with the following values:\n\n${obj.members.reduce(
+        (acc, val) => acc.concat(`- ${val.name} (\`${val.type}\`)\n`),
+        ''
+      )}`,
+      '\n'
+    );
+  }
+  /* TypeDeclaration */
+  if (kind === 219) {
+    return ''.concat(
+      `### <a name="${obj.name}">${obj.name}</a>\n\n`,
+      `One of the following values:\n\n\`${obj.type}\``,
+      '\n'
+    );
+  }
+  /* PropertySignature */
+  if (kind === 141) {
+    return '';
+  }
+  /* MethodSignature */
+  if (kind === 143) {
+    let md = `### ${obj.name ? obj.name : ''}\n\n`;
+    md += `Method takes the following params:\n${obj.parameters.reduce(
+      (acc, val) => acc.concat(`- ${val.name} (\`${val.type}\`)\n`),
+      ''
+    )}\n`;
+    return md;
+  }
+  /* ModuleDeclaration */
+  if (kind === 221) {
+    let md = `## ${obj.name}\n\n`;
+    if (obj.members) {
+      obj.members.forEach(child => {
+        md = md.concat(getMarkdown(child, obj));
+      });
+    }
+    return md;
+  }
+  return '';
+};
+
+Promise.resolve(typedocs.generate([pathToDeclaration]))
+  .then(res =>
+    res
+      .filter(value => value.name && value.name === `"${pathToDeclaration}"`)[0]
+      .members.filter(x => x.kind === 217 || x.kind === 221)
+  )
+  .then(res => {
+    let markdown = '';
+    res.forEach(obj => {
+      markdown = markdown.concat(getMarkdown(obj));
+    });
+    return markdown;
+  })
+  .then(markdown => {
+    fs.readFile(pathToREADME, 'utf8', (err, data) => {
+      data = data.split('# API');
```

Comment:
So what I've seen other people do is like put some kind of comment in the markdown

`<-- API GOES HERE -->`

or something to prevent any potential collisions.

https://gist.github.com/jonikarppinen/47dc8c1d7ab7e911f4c9

(Example of markdown comments)